### DISCUSSION TITLE
HIP_RUNTIME_API.H cannot found

### DISCUSSION THREAD

**Discussion by qiyuelu1** [Nek5000/nekRS#535] [Compiling]
Hello, 
I encountered an error while compiling NekRS V23.0:
============================
_In file included from /u/qiyuelu1/ILComputes/nekRS/3rd_party/occa/src/occa/internal/modes/hip/kernel.hpp:5,
                 from /u/qiyuelu1/ILComputes/nekRS/3rd_party/occa/src/occa/internal/modes/hip/kernel.cpp:1:
/u/qiyuelu1/ILComputes/nekRS/3rd_party/occa/src/occa/internal/modes/hip/polyfill.hpp:7:12: fatal error: hip/hip_runtime_api.h: No such file or directory
    7 | #  include <hip/hip_runtime_api.h>
      |            ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated._
============================
And the configuration is: 
============================
Default_ backend : CUDA
CPU backend compiler: /sw/spack/delta-2022-03/apps/gcc/11.2.0-gcc-8.4.1-fxgnsyr/bin/g++ (flags: -w -O3 -g -march=native -mtune=native -ffast-math)
NVIDIA CUDA backend enabled (flags: -w -O3 -lineinfo --use_fast_math)
AMD HIP backend enabled (flags: -w -O3 -g -ffp-contract=fast -funsafe-math-optimizations -ffast-math -munsafe-fp-atomics)
GPU aware MPI support: ON_
============================
There are no such file if checking that directory. Did I miss something?

Thanks,
Qiyue Lu

**Comment by yslan** [comment]
Hi,

It looks like OCCA is trying to compile HIP while your default GPU beckend should be CUDA.  My guess is, there is a mismatch between our nrsconfig and occa's env detection.

Can you please try to make a clean build (`rm -r build`) with manually turning off HIP?
```
./nrsconfig -DENABLE_HIP=0 -DOCCA_HIP_ENABLED=0
```
